dcos_test_utils/ssh_client.py

```diff
@@ -169,7 +169,7 @@ def get_home_dir(self, host: str, port: int=22) -> str:
         """
         return self.command(host, ['pwd'], port=port).decode().strip()
 
-    @retrying.retry(wait_fixed=1000)
+    @retrying.retry(wait_fixed=1000, stop_max_attempt_number=1800)
```

Comment:
Nice catch, but 30min?? Why so long?

Reply:
It could easily take 15 minutes if the cluster is being slow on being deployed. 30 min is the double of that value to be safe 😄

Reply:
Well, now that I think of it, we are already waiting for the cluster to be ready... so I think we could further reduce it down to ex. 10 minutes.

But as I mention in the comment, we should come up with a solution that checks unrecoverable errors and don't attempt to re-try.

Reply:
You could use `retry_on_exception` and specify exactly which exceptions are worth retrying and which not.

Reply:
There is only one kind of exception thrown 😞

Reply:
And I did some research, the exit codes are not consistent between linux and mac `ssh`